Makefile

```diff
@@ -30,14 +39,44 @@ e2e: deploy
 $(TARGETS): vendor/modules.txt
 	./scripts/$@
 
-build-subctl: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg/subctl/ -name "*.go")
+clean:
+	rm -f $(BINARIES) $(CROSS_BINARIES) $(CROSS_TARBALLS)
+
+build: operator-image binaries
+
+build-cross: $(CROSS_TARBALLS)
+
+binaries: $(BINARIES)
+
+operator-image: vendor/modules.txt
+	operator-sdk build quay.io/submariner/submariner-operator:$(VERSION) --verbose
+
+bin/subctl: bin/subctl-$(VERSION)-$(GOOS)-$(GOARCH)$(GOEXE)
+	ln -sf $(<F) $@
+
+dist/subctl-%.tar.xz: bin/subctl-%
+	mkdir -p dist
+	tar -cJf $@ --transform "s/^bin/subctl-$(VERSION)/" $<
+
+# Versions may include hyphens so it's easier to use $(VERSION) than to extract them from the target
+bin/subctl-%: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg/subctl/ -name "*.go") vendor/modules.txt
+	mkdir -p bin
+	GOOS=$(word $(guile (- $(words $(subst -, ,$@)) 1)),$(subst -, ,$@)); \
+	GOARCH=$(word 1,$(subst ., ,$(lastword $(subst -, ,$@)))); \
+	GOEXE=$(suffix $@); \
+	export GOARCH GOEXE GOOS; \
+	$(SCRIPTS_DIR)/compile.sh \
+		--ldflags "-X github.com/submariner-io/submariner-operator/pkg/version.Version=$(VERSION) -X github.com/submariner-io/submariner-operator/pkg/version.GitRef=$(GIT_REF)" \
+		--noupx $@ ./pkg/subctl/main.go
+
+ci: generate-embeddedyamls validate test build
 
-bin/subctl: build-subctl
+generate-embeddedyamls: pkg/subctl/operator/common/embeddedyamls/yamls.go
```

Comment:
Could this be a Makefile variable instead? Then you could use it as a dependency of `bin/subctl-%: `

Reply:
This is only here to allow other pieces of the builds to continue working as-is. It will be removed when `test` and `validate` are pulled into the Makefile too...